contextual_orchestrator/provider_catalog.py

```
@@ -0,0 +1,960 @@
+            provider_name=account.provider_name,
+        )
+        ModelClient(timeout=max(1, int(self.timeout_seconds)))._validate_provider(probe)
+        url = account.models_url or ""
+        request = urllib.request.Request(
+            url,
+            headers={
+                account.auth_header_name: f"{account.auth_prefix} {credential}".strip(),
+                "Accept": "application/json",
+            },
+            method="GET",
+        )
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)